sorting/dnf_sort.c

```diff
@@ -0,0 +1,101 @@
+/**
+ * @file
+ * @brief Program to perform [dutch national
+ * flag](https://en.wikipedia.org/wiki/Dutch_national_flag_problem) of a target
+ * value in a given {0, 1, 2} containing array.
+ * @authors [Bijan Riesenberg](https://github.com/Data-Hero) Iterative
+ */
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+/**
+ * @brief Dutch National Flag 3-Way inplace sort algorithm
+ * @param arr array containting elements from {0, 1, 2} to be sorted
+ * @param size size of the array
+ */
+void dnfSort(int8_t *arr, int size)
+{
+    int low = 0, mid = 0, high = size - 1;
+    int8_t temp = 0;
+    while (mid <= high)
+    {
+        switch (arr[mid])
+        {
+        case 0:
+            temp = arr[low];
+            arr[low++] = arr[mid];
+            arr[mid++] = temp;
+            break;
+        case 1:
+            ++mid;
+            break;
+        case 2:
+            temp = arr[mid];
+            arr[mid++] = arr[high];
+            arr[high--] = temp;
+            break;
+        default:
+            break;
+        }
+    }
+}
+
+/**
+ * @brief Print an array containing int8_t elements
+ *
+ * @param arr array containting elements from {0, 1, 2} (int8_t) to be printed
+ * @param size size of the array
+ */
+void printArray(int8_t *arr, int size)
+{
+    for (int c = 0; c < size; c++)
+    {
+        c != size - 1 ? printf("%" PRIi8 ", ", arr[c])
+                      : printf("%" PRIi8 "\n", arr[c]);
+    }
+}
+
```

Comment:
Please add self-test cases so we can verify your program works properly (**see the [typical structure of a program](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) as a reference**). Let us know if you need any help here or in our [**Discord server**](https://discord.gg/c7MnfGFGa6). 😃